libminifi/src/RemoteProcessorGroupPort.cpp

```diff
@@ -76,27 +89,20 @@ void RemoteProcessorGroupPort::returnProtocol(std::unique_ptr<Site2SiteClientPro
 void RemoteProcessorGroupPort::initialize() {
 // Set the supported properties
   std::set<core::Property> properties;
-  properties.insert(hostName);
-  properties.insert(port);
   properties.insert(portUUID);
   setSupportedProperties(properties);
 // Set the supported relationships
   std::set<core::Relationship> relationships;
   relationships.insert(relation);
   setSupportedRelationships(relationships);
+  curl_global_init(CURL_GLOBAL_DEFAULT);
```

Comment:
the function is not thread safe. initialization isn't guaranteed to be thread safe either since other components could initialize concurrently. It's probably fine for now but seems like this should be done in a thread safe way. 
https://curl.haxx.se/libcurl/c/curl_global_init.html